### PR TITLE
Make BaseTypeNullable a transitive check.

### PR DESCRIPTION
**Pull request by dekimir**

### CODE REVIEW

**Comment by dekimir** [comment]
@umar456 PTAL

File: source/validate_id.cpp [hunk 1]
```diff
@@ -609,64 +609,54 @@ bool idUsage::isValid<SpvOpConstantSampler>(const spv_instruction_t* inst,
   return true;
 }
 
+bool IsBaseTypeNullable(const std::vector<uint32_t>& instruction,
```
Comment (dneto0) [inline source/validate_id.cpp]: Missing a method blurb.
In particular, it's not immediately clear why you'd need a the usedefs.
Also, should say that you expect the instruction to be a type-defining instruction, and you're checking that the type being defined is what is nullable.

Comment (dneto0) [inline source/validate_id.cpp]: Also, why is it "Base" type?

**Comment by dneto0** [comment]
-1 for the missing blurb.  Potentially also for the "Base" part of the name, which doesn't seem necessary.

Optional: Add a negative test case for OpTypeRuntimeArray.

**Comment by dekimir** [comment]
Renamed to `IsTypeNullable`, added a blurb, and added a neg test for `OpTypeRuntimeArray`.

**Comment by dneto0** [comment]
+2
I'll squash and merge.

**Comment by dekimir** [comment]
Also clarified the error message a bit.

**Pull request merged**